File: CadastroScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet } from 'react-native';

export default function CadastroScreen({ navigation }) {
  const [nome, setNome] = useState('');
  const [email, setEmail] = useState('');
  const [senha, setSenha] = useState('');
  const [confirmarSenha, setConfirmarSenha] = useState('');
  const [erro, setErro] = useState('');

  const handleCadastro = () => {
    if (!nome || !email || !senha || !confirmarSenha) {
      setErro('Preencha todos os campos.');
      return;
    }

    if (senha.length < 6) {
      setErro('A senha deve ter pelo menos 6 caracteres.');
      return;
    }

    if (senha !== confirmarSenha) {
      setErro('Senhas não combinam.');
      return;
    }

    setErro('');
    navigation.navigate('BoasVindas', {
      origem: 'cadastro',
      usuario: { nome },
    });
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Criar Conta</Text>

      <TextInput
        style={styles.input}
        placeholder="Nome"
        placeholderTextColor="#EFD6AC"
        value={nome}
        onChangeText={setNome}
      />

      <TextInput
        style={styles.input}
        placeholder="Email"
        placeholderTextColor="#EFD6AC"
        value={email}
        onChangeText={setEmail}
      />

      <TextInput
        style={styles.input}
        placeholder="Senha"
        placeholderTextColor="#EFD6AC"
        secureTextEntry
        value={senha}
        onChangeText={setSenha}
      />

      <TextInput
        style={styles.input}
        placeholder="Confirmar senha"
        placeholderTextColor="#EFD6AC"
        secureTextEntry
        value={confirmarSenha}
        onChangeText={setConfirmarSenha}
      />

      {erro !== '' && <Text style={styles.error}>{erro}</Text>}

      <TouchableOpacity style={styles.button} onPress={handleCadastro}>
        <Text style={styles.buttonText}>Cadastrar</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#04151F',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 30,
  },
  title: {
    fontSize: 32,
    color: '#EFD6AC',
    fontWeight: 'bold',
    marginBottom: 40,
    fontFamily: 'serif',
  },
  input: {
    width: '100%',
    backgroundColor: '#183A37',
    color: '#EFD6AC',
    padding: 14,
    borderRadius: 8,
    marginBottom: 20,
    fontSize: 16,
  },
  error: {
    color: '#FF6F61',
    fontSize: 14,
    fontStyle: 'italic',
    marginBottom: 20,
    textAlign: 'center',
  },
  button: {
    backgroundColor: '#C44900',
    paddingVertical: 14,
    paddingHorizontal: 40,
    borderRadius: 8,
    borderWidth: 1,
    borderColor: '#432534',
    shadowColor: '#000',
    shadowOpacity: 0.3,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
  },
  buttonText: {
    color: '#04151F',
    fontSize: 16,
    fontWeight: 'bold',
    letterSpacing: 1,
    textAlign: 'center',
  },
});

File: orpheusapp/lib/CadastroScreen.dart
import 'package:flutter/material.dart';
// ignore: unused_import
import 'package:orpheusapp/database.dart';  // <-- caminho correto
import 'LoginScreen.dart';

class CadastroScreen extends StatefulWidget {
  const CadastroScreen({super.key});

  @override
  State<CadastroScreen> createState() => _CadastroScreenState();
}

class _CadastroScreenState extends State<CadastroScreen> {
  String nome = "";
  String email = "";
  String senha = "";
  String confirmarSenha = "";
  String erro = "";

  // --- Função de cadastro ---
  Future<void> handleCadastro() async {
    // Validações básicas
    if (nome.isEmpty || email.isEmpty || senha.isEmpty || confirmarSenha.isEmpty) {
      setState(() => erro = "Preencha todos os campos.");
      return;
    }

    if (senha.length < 6) {
      setState(() => erro = "A senha deve ter pelo menos 6 caracteres.");
      return;
    }

    if (senha != confirmarSenha) {
      setState(() => erro = "Senhas não combinam.");
      return;
    }

    setState(() => erro = "");

    try {
      // Salvando email como username (igual ao seu RN)
      await DatabaseHelper.registerUser(email, senha);

      // ALERTA
      if (!mounted) return;
      showDialog(
        context: context,
        builder: (_) => AlertDialog(
          title: const Text("Conta Criada!"),
          content: Text("Bem-vindo(a), $nome. Faça login para continuar."),
          actions: [
            TextButton(
              onPressed: () {
                Navigator.pop(context); // Fecha alerta
                Navigator.pushReplacement(
                  context,
                  MaterialPageRoute(builder: (context) => const LoginScreen()),
                );
              },
              child: const Text("Ir para Login"),
            ),
          ],
        ),
      );
    } catch (e) {
      setState(() => erro = "Erro ao cadastrar. Este email pode já estar em uso.");
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF04151F),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(30),
        child: Center(
          child: Column(
            children: [

              const SizedBox(height: 40),

              const Text(
                "Criar Conta",
                style: TextStyle(
                  fontSize: 32,
                  color: Color(0xFFEFD6AC),
                  fontWeight: FontWeight.bold,
                  fontFamily: "serif",
                ),
              ),

              const SizedBox(height: 40),

              // ---------------------
              // CAMPOS DE ENTRADA
              // ---------------------

              _buildInput(
                label: "Nome",
                onChanged: (v) => setState(() => nome = v),
              ),

              _buildInput(
                label: "Email",
                onChanged: (v) => setState(() => email = v),
                keyboard: TextInputType.emailAddress,
              ),

              _buildInput(
                label: "Senha",
                onChanged: (v) => setState(() => senha = v),
                isPassword: true,
              ),

              _buildInput(
                label: "Confirmar senha",
                onChanged: (v) => setState(() => confirmarSenha = v),
                isPassword: true,
              ),

              // MENSAGEM DE ERRO
              if (erro.isNotEmpty)
                Padding(
                  padding: const EdgeInsets.only(bottom: 20),
                  child: Text(
                    erro,
                    style: const TextStyle(
                      color: Color(0xFFFF6F61),
                      fontSize: 14,
                      fontStyle: FontStyle.italic,
                    ),
                    textAlign: TextAlign.center,
                  ),
                ),

              // BOTÃO CADASTRAR
              ElevatedButton(
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFFC44900),
                  padding: const EdgeInsets.symmetric(vertical: 14),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(8),
                    side: const BorderSide(
                      color: Color(0xFF432534),
                      width: 1,
                    ),
                  ),
                  elevation: 4,
                  minimumSize: const Size(double.infinity, 50),
                ),
                onPressed: handleCadastro,
                child: const Text(
                  "Cadastrar",
                  style: TextStyle(
                    color: Color(0xFF04151F),
                    fontSize: 16,
                    fontWeight: FontWeight.bold,
                    letterSpacing: 1,
                  ),
                ),
              ),

              const SizedBox(height: 20),

              // BOTÃO VOLTAR
              GestureDetector(
                onTap: () => Navigator.pop(context),
                child: const Text(
                  "Voltar",
                  style: TextStyle(
                    color: Color(0xFFEFD6AC),
                    decoration: TextDecoration.underline,
                  ),
                ),
              ),

              const SizedBox(height: 40),
            ],
          ),
        ),
      ),
    );
  }

  // -------------------------
  //   WIDGET DE INPUT
  // -------------------------
  Widget _buildInput({
    required String label,
    required void Function(String) onChanged,
    bool isPassword = false,
    TextInputType keyboard = TextInputType.text,
  }) {
    return Container(
      margin: const EdgeInsets.only(bottom: 20),
      child: TextField(
        onChanged: onChanged,
        keyboardType: keyboard,
        obscureText: isPassword,
        style: const TextStyle(
          color: Color(0xFFEFD6AC),
          fontSize: 16,
        ),
        decoration: InputDecoration(
          hintText: label,
          hintStyle: const TextStyle(color: Color(0xFFEFD6AC)),
          fillColor: const Color(0xFF183A37),
          filled: true,
          contentPadding: const EdgeInsets.all(14),
          border: OutlineInputBorder(
            borderRadius: BorderRadius.circular(8),
            borderSide: BorderSide.none,
          ),
        ),
      ),
    );
  }
}
